File: scripts/analyze_content.py

import os
import json
import requests
from datetime import datetime

# Load env manually to avoid dependency issues
env_vars = {}
try:
    with open('.env', 'r') as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#') and '=' in line:
                key, value = line.split('=', 1)
                env_vars[key] = value.strip('"').strip("'")
except Exception as e:
    print(f"Error reading .env: {e}")

SUPABASE_URL = env_vars.get('SUPABASE_URL')
SUPABASE_KEY = env_vars.get('SUPABASE_SERVICE_ROLE_KEY')

if not SUPABASE_URL or not SUPABASE_KEY:
    print("Missing Supabase credentials")
    exit(1)

def analyze_content():
    print("🔍 Starting Content Analysis (Python)...")
    
    headers = {
        "apikey": SUPABASE_KEY,
        "Authorization": f"Bearer {SUPABASE_KEY}",
        "Content-Type": "application/json"
    }
    
    # Fetch carousels via REST API
    url = f"{SUPABASE_URL}/rest/v1/linkedin_carousels?select=*&order=created_at.asc"
    
    try:
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        carousels = response.json()
    except Exception as e:
        print(f"Error fetching data: {e}")
        return

    print(f"📊 Found {len(carousels)} carousels.")

    report = {
        "total": len(carousels),
        "by_status": {},
        "by_profile": {},
        "data_quality_issues": [],
        "first_post_analysis": None
    }

    for index, c in enumerate(carousels):
        # Status
        status = c.get('status', 'unknown')
        report['by_status'][status] = report['by_status'].get(status, 0) + 1
        
        # Profile
        profile = c.get('profile_type', 'unknown')
        report['by_profile'][profile] = report['by_profile'].get(profile, 0) + 1
        
        # Quality
        slides = c.get('slides')
        images = c.get('image_urls')
        
        if not slides or (isinstance(slides, list) and len(slides) == 0):
            report['data_quality_issues'].append(f"ID {c.get('id')}: Missing slides")
            
        if not images or (isinstance(images, list) and len(images) == 0):
            report['data_quality_issues'].append(f"ID {c.get('id')}: Missing image_urls")
            
        # First Post Analysis
        if index == 0:
            caption = c.get('caption', '')
            report['first_post_analysis'] = {
                "id": c.get('id'),
                "topic": c.get('topic'),
                "created_at": c.get('created_at'),
                "slides_count": len(slides) if isinstance(slides, list) else 0,
                "image_count": len(images) if isinstance(images, list) else 0,
                "caption_preview": (caption[:100] + '...') if caption else 'N/A',
                "raw_data": c
            }

    # Output Report
    execution_dir = os.path.join(os.getcwd(), 'execution')
    if not os.path.exists(execution_dir):
        os.makedirs(execution_dir)
        
    report_path = os.path.join(execution_dir, 'CONTENT_AUDIT.md')
    
    md_content = f"""# Content Audit Report

**Date:** {datetime.now().isoformat()}
**Total Records:** {report['total']}

## 📊 Distribution
### By Status
```json
{json.dumps(report['by_status'], indent=2)}
```

### By Profile Type
```json
{json.dumps(report['by_profile'], indent=2)}
```

## 🚨 Data Quality Issues
{chr(10).join(['- ' + i for i in report['data_quality_issues']]) if report['data_quality_issues'] else "No major structure issues found."}

## 🔍 First Post Analysis
**ID:** `{report['first_post_analysis']['id'] if report['first_post_analysis'] else 'N/A'}`
**Topic:** {report['first_post_analysis']['topic'] if report['first_post_analysis'] else 'N/A'}
**Created:** {report['first_post_analysis']['created_at'] if report['first_post_analysis'] else 'N/A'}

### Issues Detected (Automated)
- [{'x' if report['first_post_analysis'] and report['first_post_analysis']['image_count'] > 0 else ' '}] Has Images (Count: {report['first_post_analysis']['image_count'] if report['first_post_analysis'] else 0})
- [{'x' if report['first_post_analysis'] and report['first_post_analysis']['slides_count'] > 0 else ' '}] Has Slides (Count: {report['first_post_analysis']['slides_count'] if report['first_post_analysis'] else 0})

### Raw Record (First Post)
```json
{json.dumps(report['first_post_analysis']['raw_data'] if report['first_post_analysis'] else {}, indent=2)}
```
"""

    with open(report_path, 'w') as f:
        f.write(md_content)
    
    print(f"✅ Audit report saved to: {report_path}")

if __name__ == "__main__":
    analyze_content()

File: scripts/legacy/analyze_content.ts

import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import fs from 'fs';
import path from 'path';

dotenv.config();

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('Missing Supabase credentials');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function analyzeContent() {
  console.log('🔍 Starting Content Analysis...');

  const { data: carousels, error } = await supabase
    .from('linkedin_carousels')
    .select('*')
    .order('created_at', { ascending: true });

  if (error) {
    console.error('Error fetching carousels:', error);
    return;
  }

  console.log(`📊 Found ${carousels.length} carousels (including single-image posts).`);

  const report = {
    total: carousels.length,
    by_status: {} as Record<string, number>,
    by_profile: {} as Record<string, number>,
    data_quality_issues: [] as string[],
    first_post_analysis: null as any
  };

  carousels.forEach((c, index) => {
    // Status counts
    const status = c.status || 'unknown';
    report.by_status[status] = (report.by_status[status] || 0) + 1;

    // Profile counts
    const profile = c.profile_type || 'unknown';
    report.by_profile[profile] = (report.by_profile[profile] || 0) + 1;

    // Data Quality Checks
    if (!c.slides || (Array.isArray(c.slides) && c.slides.length === 0)) {
        report.data_quality_issues.push(`ID ${c.id}: Missing slides`);
    }
    if (!c.image_urls || (Array.isArray(c.image_urls) && c.image_urls.length === 0)) {
        report.data_quality_issues.push(`ID ${c.id}: Missing image_urls`);
    }

    // First Post Analysis (index 0)
    if (index === 0) {
        report.first_post_analysis = {
            id: c.id,
            topic: c.topic,
            created_at: c.created_at,
            slides_count: c.slides?.length,
            image_count: c.image_urls?.length,
            caption_preview: c.caption?.substring(0, 100) + '...',
            raw_data: c
        };
    }
  });

  // Output Report
  const reportPath = path.join(process.cwd(), 'execution', 'CONTENT_AUDIT.md');
  const mdContent = `# Content Audit Report

**Date:** ${new Date().toISOString()}
**Total Records:** ${report.total}

## 📊 Distribution
### By Status
\`\`\`json
${JSON.stringify(report.by_status, null, 2)}
\`\`\`

### By Profile Type
\`\`\`json
${JSON.stringify(report.by_profile, null, 2)}
\`\`\`

## 🚨 Data Quality Issues
${report.data_quality_issues.length > 0 ? report.data_quality_issues.map(i => `- ${i}`).join('\n') : "No major structure issues found."}

## 🔍 First Post Analysis
**ID:** \`${report.first_post_analysis?.id}\`
**Topic:** ${report.first_post_analysis?.topic}
**Created:** ${report.first_post_analysis?.created_at}

### Issues Detected (Automated)
- [${report.first_post_analysis?.image_count > 0 ? 'x' : ' '}] Has Images
- [${report.first_post_analysis?.slides_count > 0 ? 'x' : ' '}] Has Slides

### Raw Record (First Post)
\`\`\`json
${JSON.stringify(report.first_post_analysis?.raw_data, null, 2)}
\`\`\`
`;

  fs.writeFileSync(reportPath, mdContent);
  console.log(`✅ Audit report saved to: ${reportPath}`);
}

analyzeContent().catch(console.error);
